shell/browser/api/atom_api_web_contents.cc

```diff
@@ -894,7 +894,15 @@ void WebContents::RenderViewDeleted(content::RenderViewHost* render_view_host) {
 }
 
 void WebContents::RenderProcessGone(base::TerminationStatus status) {
-  Emit("crashed", status == base::TERMINATION_STATUS_PROCESS_WAS_KILLED);
+  // Emit the crashed event in the next tick.
+  base::PostTask(
+      FROM_HERE, {content::BrowserThread::UI},
+      base::BindOnce(
+          [](base::WeakPtr<WebContents> self, bool killed) {
+            if (self)
+              self->Emit("crashed", killed);
```

Comment:
Does this introduce a race condition where the WebContents might disappear before the 'crash' event is emitted?

Reply:
I agree with @nornagon in theory there is a race condition here.  Could we instead emit `crashed` in JS inside a `nextTick` or `setTimeout`.  Then we can guarantee lifetimes and no lost events